assets/js/components/surveys/CurrentSurvey.js

```diff
@@ -16,32 +16,173 @@
  * limitations under the License.
  */
 
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useState } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
-import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { CORE_FORMS } from '../../googlesitekit/datastore/forms/constants';
 import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
-import Survey from './Survey';
-import Portal from '../Portal';
-const { useSelect } = Data;
+import SurveyCompletion from './SurveyCompletion';
+import SurveyQuestionRating from './SurveyQuestionRating';
+const { useDispatch, useSelect } = Data;
+
+const ComponentMap = {
+	rating: SurveyQuestionRating,
+};
 
 const CurrentSurvey = () => {
-	const usingProxy = useSelect( ( select ) => select( CORE_SITE ).isUsingProxy() );
-	const currentSurvey = useSelect( ( select ) => select( CORE_USER ).getCurrentSurvey() );
+	const [ hasSentSurveyShownEvent, setHasSentSurveyShownEvent ] = useState( false );
+	const [ hasSentCompletionEvent, setHasSentCompletionEvent ] = useState( false );
+
+	useEffect( () => {
+		if ( questions?.length && ! hasSentSurveyShownEvent ) {
+			setHasSentSurveyShownEvent( true );
+			sendSurveyEvent( 'survey_shown' );
+		}
+	}, [ questions, hasSentSurveyShownEvent, sendSurveyEvent ] );
```

Comment:
`sendSurveyEvent` is `undefined` here I think, it's only assigned to a variable further below.